src/olympia/addons/indexers.py

```diff
@@ -388,3 +393,120 @@ def extract_document(cls, obj):
                     cls.extract_field_raw_translations(preview, 'caption'))
 
         return data
+
```

Comment:
Everything here is copied from `src/olympia/search/indexers.py` which I deleted. `reindex_tasks_group()` is the ne and improved `reindex()` which instead of firing reindex tasks directly, just returns a group of signatures via `create_chunked_tasks_signatures()`.